src/clj/puppetlabs/http/client/common.clj

```diff
@@ -149,3 +155,28 @@
 
 (def Response
   (schema/either NormalResponse ErrorResponse))
+
+(def OptionalMetricRegistry
+  (schema/maybe MetricRegistry))
+
+(def Metrics
+  {schema/Str Timer})
+
+(def MetricData
+  {:metric-id schema/Str
+   :count schema/Int
+   :mean schema/Num
+   :aggregate schema/Num})
+
+(def MetricsData
+  {schema/Str MetricData})
+
+(def MetricFilter
+  (schema/conditional
+   #(contains? % :url)
+   {:metric-type (schema/enum "bytes-read")
```

Comment:
+1 on that.  Ideally I'd like to figure out a way to get rid of the `contains?` stuff in these, as it feels fragile.  That said, as long as it's not public API, I probably wouldn't block the PR on it.